File: src/main/java/cabalgatas_salento/reservas/repository/ReservacionRepository.java
package cabalgatas_salento.reservas.repository;

import cabalgatas_salento.reservas.entity.Reservacion;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;

public interface ReservacionRepository extends JpaRepository<Reservacion, Long> {

    List<Reservacion> findByClienteId(Long clienteId);

    List<Reservacion> findBySalidaId(Long salidaId);

    @Query(nativeQuery = true, value = """
            SELECT COALESCE(SUM(num_people), 0)
            FROM reservaciones
            WHERE salida_id = :salidaId AND estado != 'cancelado'
            """)
    int sumarPersonasPorSalida(@Param("salidaId") Long salidaId);
}

File: src/reservaciones/repository/reservacion.repository.ts
import { Reservacion } from '../entities/reservacion.entity';

export abstract class ReservacionRepository {
  abstract findById(id: number): Promise<Reservacion | null>;
  abstract findByCliente(clienteId: number): Promise<Reservacion[]>;
  abstract sumarPersonasPorSalida(salidaId: number): Promise<number>;
  abstract save(reservacion: Reservacion): Promise<Reservacion>;
}
